rspace/src/test/scala/coop/rchain/rspace/history/RadixTreeSpec.scala

```diff
@@ -0,0 +1,726 @@
+          "      [20]LEAF: prefix = 13, data = 0000...0011",
+          "      [22]LEAF: prefix = 25, data = 0000...0055"
+        )
+
+        _ = printedTree1 shouldBe etalonTree1
+        _ = printedTree2 shouldBe etalonTree2
+      } yield ()
+  }
+  "appending leaf to leaf" should "create node with two leafs" in withImplAndStore { (impl, _) =>
+    val keys = Vector[Seq[Byte]](
+      TestData.hexKey("00000000"),
+      TestData.hexKey("34564544")
+    )
+    for {
+      rootItem1Opt <- impl.update(
+                       RadixTree.EmptyItem,
+                       keys(0),
+                       createBV32(0x11.toByte)
+                     )
+
+      rootItem2Opt <- impl.update(
+                       rootItem1Opt.get,
+                       keys(1),
+                       createBV32(0x55.toByte)
+                     )
+
+      rootNode <- impl.constructNodeFromItem(rootItem2Opt.get)
+
+      printedTreeStr <- impl.printTree(rootNode, "TREE: TWO LEAFS", true)
+
+      etalonTree = Vector(
+        "TREE: TWO LEAFS: root =>",
+        "   [00]LEAF: prefix = 000000, data = 0000...0011",
+        "   [34]LEAF: prefix = 564544, data = 0000...0055"
+      )
+
+      _ = printedTreeStr shouldBe etalonTree
+    } yield ()
+  }
+
+  "updating leaf" should "work correctly" in withImplAndStore { (impl, _) =>
+    val firstLeafData = createBV32(0xCB.toByte)
+    val newLeafData   = createBV32(0xFF.toByte)
+    val leafKey       = TestData.hexKey("0123456F1").toVector
+    for {
+
+      //  Create tree with one leaf
+      item1Opt <- impl.update(RadixTree.EmptyItem, leafKey, firstLeafData)
+
+      rootNode1 <- impl.constructNodeFromItem(item1Opt.get)
+      printedTree1 <- impl.printTree(
+                       rootNode1,
+                       "TREE WITH ONE LEAF",
+                       true
+                     )
+
+      item2Opt  <- impl.update(item1Opt.get, leafKey, newLeafData)
+      itemIdx   <- Sync[Task].delay(byteToInt(leafKey.head))
+      rootNode2 <- impl.constructNodeFromItem(item2Opt.get)
+
+      printedTree2 <- impl.printTree(
+                       rootNode2,
+                       "TREE WITH ONE LEAF (AFTER CHANGING DATA)",
+                       true
+                     )
+
+      etalonTree1 = Vector(
+        "TREE WITH ONE LEAF: root =>",
+        "   [00]LEAF: prefix = 123456F1, data = 0000...00CB"
+      )
+
+      etalonTree2 = Vector(
+        "TREE WITH ONE LEAF (AFTER CHANGING DATA): root =>",
+        "   [00]LEAF: prefix = 123456F1, data = 0000...00FF"
+      )
+
+      _ = printedTree1 shouldBe etalonTree1
+      _ = printedTree2 shouldBe etalonTree2
+    } yield ()
+  }
+
+  "RadixTreeImpl" should "not allow to enter keys with different lengths" in withImplAndStore {
+    (impl, _) =>
+      val leafData    = createBV32(0xCB.toByte)
+      val leafKey     = TestData.hexKey("0123456F1").toVector
+      val testLeafKey = TestData.hexKey("112").toVector
```

Comment:
@hilltracer Why is not possible to insert keys with different length if they are in different sub-root?

Reply:
It's possible. We have wrong Assertion error message. @maximus19910112  please rewrite this message to `RadixKey must not be a complete part of another radixKey in this tree.` In your unit tests and in Radix - implementation.
Also please rewrite this message In a similar way: https://github.com/hilltracer/rchain/blob/3991334c48ab79aa8b515896f26b74f6d85d12fc/rspace/src/main/scala/coop/rchain/rspace/history/RadixTree.scala#L890